test/fungibletoken.test.js

```diff
@@ -0,0 +1,360 @@
+        name: "switchboard/add_vault_capability",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Third step: remove vault capability
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/remove_vault_capability",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    // Fourth step: verify that capabilities are returned
+    const [result, e] = await executeScript({
+      code: get_vault_capabilities,
+      args: [fungibleTokenSwitchboardUser]
+    });
+    expect(result).toStrictEqual({});
+  });
+
+  // Fourth test checks if switchboard user is able to receive ft through the
+  // switchboard deposit function
+  test("should be able to receive tokens through switchboard", async () => {
+    //First step: setup switchboard
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/setup_account",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Second step: setup example token vault
+    await shallPass(
+      sendTransaction({
+        name: "setup_account",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Third step: add vault capability
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/add_vault_capability",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Fourth step: mint tokens into service account
+    await shallPass(
+      sendTransaction({
+        name: "mint_tokens",
+        args: [serviceAccount, 100],
+        signers: [serviceAccount]
+      })
+    );
+    //Fifth step: transfer tokens to switchboard user
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/transfer_tokens",
+        args: [fungibleTokenSwitchboardUser, 50],
+        signers: [serviceAccount]
+      })
+    );
+  });
+
+  // Fifth test checks if vault capabilities could be retrieved from a switchboard
+  test("should be able to retrieve vault capabilities", async () => {
+    //First step: setup switchboard
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/setup_account",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Second step: setup example token vault
+    await shallPass(
+      sendTransaction({
+        name: "setup_account",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    //Third step: add vault capability
+    await shallPass(
+      sendTransaction({
+        name: "switchboard/add_vault_capability",
+        args: [],
+        signers: [fungibleTokenSwitchboardUser]
+      })
+    );
+    // Fourth step: verify that capabilities are returned
+    const [result, e] = await executeScript({
+      code: get_vault_capabilities,
+      args: [fungibleTokenSwitchboardUser]
+    });
+    expect(result).not.toBe(null);
+  });
+  
+});
```

Comment:
Maybe add a bad test as well to validate that contract would revert the transaction when non-added vault funds are transferred.

Reply:
that will requiere to add a new example token "2" or some other token to the repo, but I really like the idea of checking that. @joshuahannan any thoughts?